test/controllers/images_controller_test.rb

```diff
@@ -109,4 +111,23 @@ def test_should_show_image_with_tags_alphabetically
       end
     end
   end
+
+  def test_should_delete_image_with_flash_message
+    testimage = Image.create!(imageurl: 'http://abc.png', tag_list: 'c, b, a')
+
+    assert_difference('Image.count', -1) do
+      delete image_path(testimage)
+    end
+    assert_redirected_to images_path
+    assert_equal 'You have successfully deleted the image.', flash[:success]
+  end
+
+  def test_should_delete_image_fails
+    # Try deleting an image ID that does not exist
+    assert_difference('Image.count', 0) do
+      delete image_path(2)
```

Comment:
A common idiom here is to use the id `-1`, because we know that such an ID will never actually be in the database. If we use `2`, this test may fail if someone adds a fixture image, and you might pull your hair out trying to figure out why it's failing 😂